json-schemas/definitions.json

```diff
@@ -14,6 +14,14 @@
     "did": {
       "type": "string",
       "pattern": "^did:([a-z0-9]+):((?:(?:[a-zA-Z0-9._-]|(?:%[0-9a-fA-F]{2}))*:)*((?:[a-zA-Z0-9._-]|(?:%[0-9a-fA-F]{2}))+))((;[a-zA-Z0-9_.:%-]+=[a-zA-Z0-9_.:%-]*)*)(\/[^#?]*)?([?][^#]*)?(#.*)?$"
+    },
+    "dateCreated": {
+      "type": "string",
+      "pattern": "^(\\d{4})-(\\d{2})-(\\d{2})T(\\d{2}):(\\d{2}):(\\d{2}(?:\\.\\d*)?)((-(\\d{2}):(\\d{2})|Z)?)$"
```

Comment:
Will need to add tests to prove the code works (as seen in [here](https://github.com/TBD54566975/dwn-sdk-js/blob/ae3d7aeb2758f0547578b62fa481aad6b961f8ef/tests/interfaces/collections/messages/collections-write.spec.ts)), I'd imagine you'll need to fix a few regressed tests also since such check didn't exist before.